### DISCUSSION TITLE
Any stable API?

### DISCUSSION THREAD

**Discussion by KrySeyt** [vrtmrz/obsidian-livesync#458] [Q&A]
I wanna integrate obsidian notes with Live Sync to my notifications system. I wanna send notifications based on obsidian notes and etc. Is there any stable API or good way for getting .md files content and events like "New file created" except of just db pooling and collecting files by chunks?

**Comment by vrtmrz** [comment]
Thank you for asking!
We can choose several ways. The API of Obsidian is the most stable way ([Developer docs](https://docs.obsidian.md/Reference/TypeScript+API/Vault/on('create')) is here).

If you want to wire from the Self-hosted LiveSync, `app.plugins.plugins["obsidian-livesync"].localDatabase.localDatabase` is an instance of PouchDB. So [changes](https://pouchdb.com/api.html#changes) might be useful. However, finally, Self-hosted LiveSync will write the document and it will be detected by the API of Obsidian. Hence the Obsidian API is the most reliable and stable way.

I hope that this information will help you!
